worlds/stardew_valley/test/TestRules.py

```diff
@@ -596,6 +597,38 @@ def swap_museum_and_bathhouse(multiworld, player):
     bathhouse_entrance.connect(museum_region)
 
 
+class TestToolVanillaRequiresBlacksmith(SVTestBase):
+    options = {
+        options.EntranceRandomization: options.EntranceRandomization.option_buildings,
+        options.ToolProgression: options.ToolProgression.option_vanilla,
+    }
+
+    def test_cannot_make_get_any_tool_without_blacksmith_access(self):
+        railroad_item = "Railroad Boulder Removed"
+        swap_blacksmith_and_bathhouse(self.multiworld, self.player)
+        collect_all_except(self.multiworld, railroad_item)
+
+        for tool in [Tool.pickaxe, Tool.axe, Tool.hoe, Tool.trash_can, Tool.watering_can]:
```

Comment:
Could you add an assert that we **can** get the fishing rod? To confirm that it isn't being linked to the blacksmith like one of the bugs suggested?

Reply:
I added a different test to ensure fishing rod level 2 and 3 need the fish shop.